### ISSUE TITLE
No recipes after upgrading postgres docker container

### ISSUE THREAD

**Issue by bsanders** [TandoorRecipes/recipes#3127]
### Issue

Hello :)

tl;dr: I have users, but not recipes after a postgres docker container upgrade from 11->16.

I've been running Tandoor for a little over a year (via the docker-compose in the install docs on top of Ubuntu22.04).  In that time, my process for updating has been `docker-compose pull; docker-compose down; docker-compose up -d` and this has worked flawlessly until today.  Today after that process there was an error message in the logs along the lines of "you need to be using at least Postgres14", so I went googling and found the postgres section at https://docs.tandoor.dev/system/updating/ which I've apparently missed until now.

My prior postgres image was 11, based on recollection and the output of `docker images`?  The "only" anomaly I had during the postgres upgrade process linked above was there were a number of errors (including syntax errors) during the `Import the database export` step.  I can imagine there's been a schema change or two that I'm tripping over, though the syntax errors were surprising.

Needless to say when I docker-composed back up, things weren't working (HTTP 500).  The logs suggested that my password was incorrect.  I googled around and did a `docker exec -it b2bb982b322d psql postgres -U djangouser`, then did a `alter role djangouser with password 'PASSWORD_FROM_MY_ENV_FILE';.  After that, I was able to get to what looked like an empty Tandoor (peeking into the admin view, I have users, but it appears literally 0 of everything else).

Any suggestions?

I still have the `tandoor.sql` generated from those steps (which definitely has stuff in it), and I did `sudo mv postgresql/ postgresql.old`, so I still have that.  Hopefully I still have my recipes in some retrievable format.

### Tandoor Version

Unfortunately the /system/ page is not functioning

### OS Version

Ubuntu 22.04

### Setup

Docker / Docker-Compose

### Reverse Proxy

No reverse proxy

### Environment file

```shell
$ cat .env  | grep -v '#.*' | grep -v '^$' | grep -v POSTGRES_PASSWORD | grep -v SECRET_KEY
DEBUG=0
SQL_DEBUG=0
ALLOWED_HOSTS=*
TIMEZONE=America/Los_Angeles
DB_ENGINE=django.db.backends.postgresql
POSTGRES_HOST=db_recipes
POSTGRES_PORT=5432
POSTGRES_USER=djangouser
POSTGRES_DB=djangodb
FRACTION_PREF_DEFAULT=0
COMMENT_PREF_DEFAULT=1
SHOPPING_MIN_AUTOSYNC_INTERVAL=5
GUNICORN_MEDIA=0
REVERSE_PROXY_AUTH=0
```
```


### Docker-Compose file

```shell
version: "3"
services:
  db_recipes:
    restart: always
    image: postgres:16-alpine
    volumes:
      - ./postgresql:/var/lib/postgresql/data
    env_file:
      - ./.env

  web_recipes:
    restart: always
    image: vabene1111/recipes
    env_file:
      - ./.env
    volumes:
      - staticfiles:/opt/recipes/staticfiles
      - nginx_config:/opt/recipes/nginx/conf.d
      - ./mediafiles:/opt/recipes/mediafiles
    depends_on:
      - db_recipes

  nginx_recipes:
    image: nginx:mainline-alpine
    restart: always
    ports:
      - 8880:80
    env_file:
      - ./.env
    depends_on:
      - web_recipes
    volumes:
      - nginx_config:/etc/nginx/conf.d:ro
      - staticfiles:/static:ro
      - ./mediafiles:/media:ro

volumes:
  nginx_config:
  staticfiles:
```

### Relevant logs

```shell
Attaching to tandoor_nginx_recipes_1, tandoor_web_recipes_1, tandoor_db_recipes_1
ESC[33mnginx_recipes_1  |ESC[0m /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
ESC[33mnginx_recipes_1  |ESC[0m /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
ESC[33mnginx_recipes_1  |ESC[0m /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
ESC[33mnginx_recipes_1  |ESC[0m 10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf is not a file or does not exist
ESC[33mnginx_recipes_1  |ESC[0m /docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
ESC[33mnginx_recipes_1  |ESC[0m /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
ESC[33mnginx_recipes_1  |ESC[0m /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
ESC[33mnginx_recipes_1  |ESC[0m /docker-entrypoint.sh: Configuration complete; ready for start up
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: using the "epoll" event method
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: nginx/1.25.5
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: built by gcc 13.2.1 20231014 (Alpine 13.2.1_git20231014) 
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: OS: Linux 5.15.0-94-generic
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: start worker processes
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: start worker process 21
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: start worker process 22
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: start worker process 23
ESC[33mnginx_recipes_1  |ESC[0m 2024/04/27 01:57:15 [notice] 1#1: start worker process 24
ESC[33mnginx_recipes_1  |ESC[0m 10.10.10.121 - - [27/Apr/2024:01:57:58 +0000] "GET / HTTP/1.1" 302 0 "-" "Uptime-Kuma/1.23.11" "-"
ESC[33mnginx_recipes_1  |ESC[0m 10.10.10.121 - - [27/Apr/2024:01:57:58 +0000] "GET /search/ HTTP/1.1" 302 0 "-" "Uptime-Kuma/1.23.11" "-"
ESC[33mnginx_recipes_1  |ESC[0m 10.10.10.121 - - [27/Apr/2024:01:57:58 +0000] "GET /accounts/login/?next=/search/ HTTP/1.1" 200 7804 "-" "Uptime-Kuma/1.23.11" "-"
ESC[33mnginx_recipes_1  |ESC[0m 10.10.10.121 - - [27/Apr/2024:01:58:58 +0000] "GET / HTTP/1.1" 302 0 "-" "Uptime-Kuma/1.23.11" "-"
ESC[33mnginx_recipes_1  |ESC[0m 10.10.10.121 - - [27/Apr/2024:01:58:58 +0000] "GET /search/ HTTP/1.1" 302 0 "-" "Uptime-Kuma/1.23.11" "-"
ESC[33mnginx_recipes_1  |ESC[0m 10.10.10.121 - - [27/Apr/2024:01:58:58 +0000] "GET /accounts/login/?next=/search/ HTTP/1.1" 200 7804 "-" "Uptime-Kuma/1.23.11" "-"
ESC[32mweb_recipes_1    |ESC[0m Checking configuration...
ESC[32mweb_recipes_1    |ESC[0m Waiting for database to be ready...
ESC[32mweb_recipes_1    |ESC[0m Database is ready
ESC[32mweb_recipes_1    |ESC[0m Migrating database
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "REVERSE_PROXY_AUTH" is deprecated. Please use "REMOTE_USER_AUTH".
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "TIMEZONE" is deprecated. Please use "TZ" instead.
ESC[32mweb_recipes_1    |ESC[0m Operations to perform:
ESC[32mweb_recipes_1    |ESC[0m   Apply all migrations: account, admin, auth, authtoken, contenttypes, cookbook, oauth2_provider, sessions, sites, socialaccount
ESC[32mweb_recipes_1    |ESC[0m Running migrations:
ESC[32mweb_recipes_1    |ESC[0m   No migrations to apply.
ESC[32mweb_recipes_1    |ESC[0m Generating static files
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "REVERSE_PROXY_AUTH" is deprecated. Please use "REMOTE_USER_AUTH".
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "TIMEZONE" is deprecated. Please use "TZ" instead.
ESC[32mweb_recipes_1    |ESC[0m js-reverse file written to /opt/recipes/cookbook/static/django_js_reverse
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "REVERSE_PROXY_AUTH" is deprecated. Please use "REMOTE_USER_AUTH".
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "TIMEZONE" is deprecated. Please use "TZ" instead.
ESC[32mweb_recipes_1    |ESC[0m 
ESC[32mweb_recipes_1    |ESC[0m 1 static file copied to '/opt/recipes/staticfiles', 574 unmodified, 1185 post-processed.
ESC[32mweb_recipes_1    |ESC[0m Done
ESC[32mweb_recipes_1    |ESC[0m [2024-04-27 01:57:34 +0000] [1] [INFO] Starting gunicorn 21.2.0
ESC[32mweb_recipes_1    |ESC[0m [2024-04-27 01:57:34 +0000] [1] [INFO] Listening at: http://[::]:8080 (1)
ESC[32mweb_recipes_1    |ESC[0m [2024-04-27 01:57:34 +0000] [1] [INFO] Using worker: gthread
ESC[32mweb_recipes_1    |ESC[0m [2024-04-27 01:57:34 +0000] [16] [INFO] Booting worker with pid: 16
ESC[32mweb_recipes_1    |ESC[0m [2024-04-27 01:57:34 +0000] [17] [INFO] Booting worker with pid: 17
ESC[32mweb_recipes_1    |ESC[0m [2024-04-27 01:57:34 +0000] [18] [INFO] Booting worker with pid: 18
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "REVERSE_PROXY_AUTH" is deprecated. Please use "REMOTE_USER_AUTH".
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "TIMEZONE" is deprecated. Please use "TZ" instead.
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "REVERSE_PROXY_AUTH" is deprecated. Please use "REMOTE_USER_AUTH".
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "TIMEZONE" is deprecated. Please use "TZ" instead.
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "REVERSE_PROXY_AUTH" is deprecated. Please use "REMOTE_USER_AUTH".
ESC[32mweb_recipes_1    |ESC[0m DEPRECATION WARNING: Environment var "TIMEZONE" is deprecated. Please use "TZ" instead.
ESC[32mweb_recipes_1    |ESC[0m ::ffff:172.30.0.4 - - [26/Apr/2024:18:57:58 -0700] "GET / HTTP/1.0" 302 0 "-" "Uptime-Kuma/1.23.11"
ESC[32mweb_recipes_1    |ESC[0m ::ffff:172.30.0.4 - - [26/Apr/2024:18:57:58 -0700] "GET /search/ HTTP/1.0" 302 0 "-" "Uptime-Kuma/1.23.11"
ESC[32mweb_recipes_1    |ESC[0m ::ffff:172.30.0.4 - - [26/Apr/2024:18:57:58 -0700] "GET /accounts/login/?next=/search/ HTTP/1.0" 200 7804 "-" "Uptime-Kuma/1.23.11"
ESC[32mweb_recipes_1    |ESC[0m ::ffff:172.30.0.4 - - [26/Apr/2024:18:58:58 -0700] "GET / HTTP/1.0" 302 0 "-" "Uptime-Kuma/1.23.11"
ESC[32mweb_recipes_1    |ESC[0m ::ffff:172.30.0.4 - - [26/Apr/2024:18:58:58 -0700] "GET /search/ HTTP/1.0" 302 0 "-" "Uptime-Kuma/1.23.11"
ESC[32mweb_recipes_1    |ESC[0m ::ffff:172.30.0.4 - - [26/Apr/2024:18:58:58 -0700] "GET /accounts/login/?next=/search/ HTTP/1.0" 200 7804 "-" "Uptime-Kuma/1.23.11"
ESC[36mdb_recipes_1     |ESC[0m 
ESC[36mdb_recipes_1     |ESC[0m PostgreSQL Database directory appears to contain a database; Skipping initialization
ESC[36mdb_recipes_1     |ESC[0m 
ESC[36mdb_recipes_1     |ESC[0m 2024-04-27 01:57:13.893 UTC [1] LOG:  starting PostgreSQL 16.2 on x86_64-pc-linux-musl, compiled by gcc (Alpine 13.2.1_git20231014) 13.2.1 20231014, 64-bit
ESC[36mdb_recipes_1     |ESC[0m 2024-04-27 01:57:13.895 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
ESC[36mdb_recipes_1     |ESC[0m 2024-04-27 01:57:13.895 UTC [1] LOG:  listening on IPv6 address "::", port 5432
ESC[36mdb_recipes_1     |ESC[0m 2024-04-27 01:57:13.903 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
ESC[36mdb_recipes_1     |ESC[0m 2024-04-27 01:57:13.920 UTC [23] LOG:  database system was shut down at 2024-04-27 01:57:08 UTC
ESC[36mdb_recipes_1     |ESC[0m 2024-04-27 01:57:13.993 UTC [1] LOG:  database system is ready to accept connections
```
```

**Comment by smilerz** [comment]
What errors are you getting?

**Comment by bsanders** [comment]
Thank you for checking in.

Other than what I described above I don't have error messages.  By the time I tried correcting things the original errors were already lost to my terminal's scrollback buffer.  I don't think I can get old docker-compose logs?  It was (from memory!) a lot of `ERROR: relation "cookbook foo" already exists` and `invalid command \n`.

I've pinned my compose file at `postgres:11-alpine` and `vabene1111/recipes:1.5.6`, `docker-compose down`, then deleted the staticfiles volume and re-ran `cat ~/tandoor.sql | sudo docker exec -i {{database_container}} psql postgres -U {{djangouser}}`.  `docker-compose pull && docker-compose up -d` and things are working again.  I'm relieved and very grateful for my happy Tandoor install.

My guess is that I should have jumped to an intermediate postgres version at some point, but didn't realize I needed to manage those myself.  Suddenly jumping from 11->16 was too far and perhaps the db backup in `tandoor.sql` was using something deprecated.  It's also entirely possible I just flubbed the commands (maybe I did a `cp postgres postgres.old` instead of an `mv`...)

I can try to reproduce when I have time over the next day or so, but in the meantime, if anyone can clarify what the correct process for updating to `latest` from those two images, I'd be appreciative.

**Comment by smilerz** [comment]
Restarting the container will generate the same errors.  You have duplicate values that need corrected which will be displayed in the logs.

**Comment by somedrunkenmonkey** [comment]
Hi,

   I can confirm this as well.

   Restore of database occurred without an issue. 

   When querying DB from command line, data is present.

   Login does not accept any existing users credentials.

   Clicking login button in upper right hand corner takes me to setup screen and asks for creation of superuser. Even with that user created, no recipes are visible.

**Comment by onlybob** [comment]
This happened to me too. The website does not recognize my login as used before and creating a new account with same exact credentials as before, system doesn't recognize it.

**Comment by smilerz** [comment]
duplicate #2939
